src/shims/fs.rs

```diff
@@ -322,16 +322,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
     fn fcntl(
         &mut self,
-        fd_op: OpTy<'tcx, Tag>,
-        cmd_op: OpTy<'tcx, Tag>,
-        start_op: Option<OpTy<'tcx, Tag>>,
+        args: &[OpTy<'tcx, Tag>],
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
         this.check_no_isolation("fcntl")?;
 
-        let fd = this.read_scalar(fd_op)?.to_i32()?;
-        let cmd = this.read_scalar(cmd_op)?.to_i32()?;
+        let (fd, cmd, start) = if args.len() == 2 {
```

Comment:
This seems odd... I think this should look more like what you did for `syscall`: first just extract `cmd`, then in the `cmd` dispatch below extract the right number of arguments.

Reply:
Ah, this one has *two* mandatory arguments. So first extract `fd` and `cmd`, and then get the rest in the dispatch below.